### PR TITLE
Added an OTel smoke test

### PR DESCRIPTION
**Pull request by dianashevchenko**
## Description
This PR adds a smoke test with one simple parent-child trace scenario which can be latter reused/expanded to add W3C/128 bits specific smoke test.

This PR also adds a smoke test with Go specific distributed trace scenario, which uses Datadog OTel API and OTel Contrib Package

### CODE REVIEW

**Comment by cbeauchesne** [comment]
> To merge after https://github.com/DataDog/dd-trace-go/pull/1958 fix is merged

Setting this PR as draft.

File: tests/apm_tracing_e2e/test_otel.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+from tests.apm_tracing_e2e.test_single_span import _get_spans_submitted, _assert_msg
+from utils import context, weblog, scenarios, missing_feature, irrelevant
+
+
+@missing_feature(
+    context.weblog_variant not in "net-http",
```
Comment (lambrospetrou) [inline tests/apm_tracing_e2e/test_otel.py:6]: Should the `"net-http"` be a list/tuple instead of a string?

Comment (dianashevchenko) [reply tests/apm_tracing_e2e/test_otel.py:6]: Works both ways 🙌

**Comment by cbeauchesne** [comment]
Could you resolve the conflict? ping me once it's done, i'll do the review.

**Comment by dianashevchenko** [comment]
@cbeauchesne conflict resolved 🙌

File: utils/_context/_scenarios.py [hunk 2]
```diff
@@ -850,6 +850,8 @@ class scenarios:
 
     otel_tracing_e2e = OpenTelemetryScenario("OTEL_TRACING_E2E")
 
+    apm_tracing_e2e_otel_span = EndToEndScenario("APM_TRACING_E2E_OTEL_SPAN")
```
Comment (cbeauchesne) [inline utils/_context/_scenarios.py:853]: `EndToEndScenario` spwans weblog with datadog tracers, not open telemetry tracer. Or did I missed something ?

Comment (dianashevchenko) [reply utils/_context/_scenarios.py:853]: I was following the suite of ``` # APM tracing end-to-end scenarios```
It is an e2e test of Datadog Otel API wrapper

**Review by marcotc — APPROVED** [review]

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
net-http weblog are massively failing on the CI, could you have a look ?

File: utils/build/docker/golang/app/net-http.go [hunk 3]
```diff
@@ -134,13 +146,120 @@ func main() {
 		appsec.TrackCustomEvent(r.Context(), name, map[string]string{"metadata0": "value0", "metadata1": "value1"})
 	})
 
+	mux.HandleFunc("/e2e_otel_span", func(w http.ResponseWriter, r *http.Request) {
```
Comment (cbeauchesne) [inline utils/build/docker/golang/app/net-http.go:149]: Can you add a parapgraph in `docs/weblog/README.md` explaining what this endpoint does ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**